packages/next/src/server-build.ts

```diff
@@ -1956,14 +1956,20 @@ export async function serverBuild({
                   // Also includes separate handling for index routes which should match to /index.action.
                   // This follows the same pattern as the rewrites for .rsc files.
                   {
-                    src: `^${path.posix.join('/', entryDirectory, '/')}`,
+                    src: `^${path.posix.join('/', entryDirectory, '/?')}`,
```

Comment:
Unrelated to the core change here, but I updated this to match the same change for RSC handling in this PR, since it seems relevant to both cases:

- https://github.com/vercel/vercel/pull/11528